frontend/public/components/catalog-items.jsx

```diff
@@ -62,6 +62,12 @@ export class CatalogTileViewPage extends React.Component {
     if (items !== prevProps.items) {
       const newCategories = categorizeItems(items);
       this.setState(this.getCategoryState(activeTabs, newCategories));
+      if (this.hasActiveFilters(filters)) {
+        const filteredItems = this.filterItems(items);
+        const filteredCategories = recategorizeItems(filteredItems, newCategories);
+        this.setState(this.getCategoryState(activeTabs, filteredCategories));
+      }
+      this.setState(this.getFilterCounts(activeTabs, filters, filterCounts, newCategories));
```

Comment:
It seems like lines 64-69 should be...

```js
       const filteredItems = this.filterItems(items);
       const filteredCategories = recategorizeItems(filteredItems, newCategories);
       this.setState(this.getCategoryState(activeTabs, filteredCategories));
       this.setState(this.getFilterCounts(activeTabs, filters, filterCounts, newCategories));
```

and then `filterItems` and `recategorizeItems` should just return all items and categories if they see no filter is set. It should pretty much be a no-op. Then you're only calling setState once for category state instead of twice.

Reply:
We need to call `categorizeItems` to get new set of initial categories based on new set of items.  Initial categories prune those categories which do not have items.   `recategorizeItems` re-categorize (filtered) items into those initial categories but may return categories with 0 items, which is ok to do for filtering since we show empty state for categories with 0 items due to filtering.
I was thinking of renaming `categorizeItems` to something like `initializeCategories`, and `recategorizeItems` to just `categorizeItems` to maybe make things clearer

Reply:
It should be a no-op, right? If there is no filter, just return all items and categories.

Reply:
I'm saying calling `const newCategories = categorizeItems(items);` and `const filteredCategories = recategorizeItems(filteredItems, newCategories);` will call categorization twice.

Reply:
Right... and shouldn't `recategorizeItems` just check that there are no filters and return the existing items and categories?

Reply:
OK I see how it's written... `recategorizeItems` doesn't do any filtering itself. I think what you have is OK.